### PR TITLE
Polyfill ontrack on browsers that don't have it.

### PR DESCRIPTION
**Pull request by jan-ivar**
Firefox 47 has this natively, so time for a polyfill.

### CODE REVIEW

**Comment by jan-ivar** [comment]
> ^ 'arrow function syntax (=>)' is only available in ES6 (use esnext option).

@KaptenJansson is using exnext an option by any chance?

**Comment by fippo** [comment]
@a2sheppy pushed you, eh? Looking forward to using this. I should natively polyfill this for Edge though, onaddstream is a hack there.

**Comment by jan-ivar** [comment]
Fixes https://github.com/webrtc/adapter/issues/171

**Comment by fippo** [comment]
@samdutton maybe you know some people with strong opinions re esnext?

**Comment by jan-ivar** [comment]
PTAL

File: test/test.js [hunk 1]
```diff
@@ -1225,6 +1225,14 @@ test('Basic connection establishment', function(t) {
...
     window.testPassed = [];
     window.testFailed = [];
+    var t = {
+      ok: function(ok, msg) {
+        window[ok ? 'testPassed' : 'testFailed'].push(msg);
+      },
+      is: function(a, b, msg) {
+        this.ok((a === b), msg + ' - got ' + b);
+      }
+    };
```
Comment (fippo) [inline test/test.js:1235]: wait, this makes life easy again?!

**Comment by fippo** [comment]
i'm very much in favor of merging this!

File: test/test.js (hunk 1 above)
Comment (jan-ivar) [inline test/test.js:1235]: Look below.

**Comment by samdutton** [comment]
> samdutton https://github.com/samdutton maybe you know some people with
> strong opinions re esnext?

I think it's fine to use any syntax that's supported by the browsers we're
targeting, if that's what you mean :).

**Comment by KaptenJansson** [comment]
Re esnext, is that message from nodejs? I agree with samdutton@ as long as it supported by browsers and nodejs it's fine.

**Comment by KaptenJansson** [comment]
LGTM.

**Comment by fippo** [comment]
no, a jshint option. But why are we still using jshint/jscs even?

**Comment by samdutton** [comment]
We're using ESLint for samples — just created new issue to do the same for
adapter.js: #176 https://github.com/webrtc/adapter/issues/176.

**Comment by KaptenJansson** [comment]
Yeah I was also wondering why we weren't using eslint everywhere, I think we wanted to see how it worked out in samples first.

**Pull request merged**